### PR TITLE
Adding .ICO header to icons.

### PR DESCRIPTION
**Pull request by LarsSchlieper**
Adaptation to other icon extraction programs which add .ICO header. 
Otherwise we get partially only raw .BMP header without BITMAPFILEHEADER.

### CODE REVIEW

File: src/PeNet/PeFile.cs [hunk 1]
```diff
@@ -478,25 +478,58 @@ public bool HasValidCertChain(bool useOnlineCrl)
         ///     and collect the corresponding bytes in an array.
         /// </summary>
         /// <returns>A byte array with Icons, an empty array if no Icons are included.</returns>
-        public IEnumerable<byte[]> Icons()
+        public IEnumerable<byte[]> IconsAsRaw()
```
Comment (rngcntr) [inline src/PeNet/PeFile.cs:481]: Do we actually want to be able to export raw data without headers? Is there any value in doing so?

File: src/PeNet/Header/Resource/Icon.cs [hunk 2]
```diff
@@ -32,5 +44,128 @@ public Span<byte> AsSpan()
...
             return PeFile.AsSpan(Offset, Size);
         }
+
+        /// <summary>
+        ///     Adding .ICO-Header to the bytes of the icon image.
+        /// </summary>
+        /// <returns>Bytes of the icon image as .ICO file.</returns>
+        public byte[]? AsICO()
+        {
+            if (AssociatedGroupIconDirectoryEntry == null) return null;
```
[the remark this answers is not in the document]
Comment (LarsSchlieper) [reply src/PeNet/Header/Resource/Icon.cs:54]: If this happens frequently, we could also extract an alternate header from the .RAW.

[the remark this answers is not in the document]
Comment (rngcntr) [reply src/PeNet/Header/Resource/Icon.cs]: We can still add such a feature in case we encounter these group icons frequently

**Review by rngcntr — APPROVED** [review]

**Review by secana — APPROVED** [review]

**Comment by secana** [comment]
@LarsSchlieper looks good. Is the PR ready to be merged? If so, I can do it and release the next preview version.

**Pull request merged**